ironmon_tracker/Drawing.lua

```diff
@@ -697,6 +759,31 @@ function Drawing.relocateAnimatedPokemon()
 	end
 end
 
+---Returns the proper full image path to an asset, while checking if any overrides are necessary; or returns nil if none exists
+---@param imagePathType string A valid image path type key, as defined in Drawing.ImagePaths
+---@param value any? Optional, a value to be used to build the image file (such as a pokemonID)
+---@return string|nil
+function Drawing.getImagePath(imagePathType, value)
+	local imagePathObj = Drawing.ImagePaths[imagePathType or false]
+	if not imagePathObj then
+		return nil
+	end
+
+	-- If an override is provided for this path, use that
+	if type(imagePathObj.getOverridePath) == "function" then
+		-- If no condition is required, just use the path; otherwise first check the use-condition
+		if not imagePathObj.shouldUseOverride or (type(imagePathObj.shouldUseOverride) == "function" and imagePathObj:shouldUseOverride(value)) then
```

Comment:
Ah. The `if not imagePathObj.shouldUseOverride` should have been a simple nil check. The logic here was meant to follow this design, which I added to the comment section in the most recent commit.

- `getOverridePath`: If specified, will *always* use this override path instead of the default
- `shouldUseOverride`: If specified, will *only* use the override path if the condition defined in this function is met

Originally I required both functions to be defined, but it wanted to loosen it up for cases where you always wanted to override the file path and didn't need a condition to use it.

Reply:
ah ok, in which case there's two things with this that I'd suggest:
- `shouldUseOverride` is an unclear/unintuitive name, to me it'd suggest a boolean rather than a function (hence my initial confusion). Perhaps naming it something like `overrideConditionFunc` instead is clearer? (i.e. to indicate "this is a condition for using the override that is specifically a function)
- the nil check isn't necessary. `type(v)` will return nil if `v` is nil, therefore the `type(imagePathObj.shouldUseOverride) == "function"` is itself the nil check